File: go32/iss_tim/GetTimerSpeed.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

static uint32_t GetTimerSpeed (void) {
  TTimerStruc *t = ISS_TimersData;
  uint32_t fastest = PIT_LATENCY_MAX;
  int c;

  for (c = ISS_MaxTimers; c; c--) {
    if (t->TActive && (t->TSpeed < fastest))
      fastest = t->TSpeed;

    t++;
  }

  return fastest;
}

File: go32/iss_tim/pas/GetTimerSpeed.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

Function GetTimerSpeed : DWord;
Var Counter  : DWord;
    TFastest : DWord;
Begin
 TFastest:=PIT_LATENCY_MAX;
 For Counter:=1 To ISS_MaxTimers Do Begin
   If ISS_TimersData[Counter].TActive And
      (ISS_TimersData[Counter].TSpeed < TFastest) Then
     TFastest:=ISS_TimersData[Counter].TSpeed;
  End;
 GetTimerSpeed:=TFastest;
End;
